### PR TITLE
isisomorphic and convert_system fix

### LINKED ISSUE #1281: `convert_system(ODESystem, sys, independent_variable(sys2)` gives `ArgumentError: Cannot find the parent of ...`
it looks like from the tests that this is known: `@test_throws ArgumentError convert_system(ODESystem, sys, t)`.

I couldn't find an issue for this, what is the upstream issue that needs to be resolved for this to work?

### PR DESCRIPTION
**Pull request by anandijain**
closes #1281

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -6,6 +6,7 @@ version = "8.8.1"
 [deps]
 AbstractTrees = "1520ce14-60c1-5f80-bbc7-55ef81b5835c"
 ArrayInterface = "4fba245c-0d91-5ea0-9b3e-6abc04ee57a9"
+Combinatorics = "861a8166-3701-5b0c-9a16-15d98fcdc6aa"
```
Comment (YingboMa) [inline Project.toml:9]: compat

File: src/systems/diffeqs/abstractodesystem.jl [hunk 2]
```diff
@@ -900,3 +900,53 @@ end
...
+end
+
+function _match_eqs(eqs1, eqs2)
+    eqpairs = Pair[]
+    for (i, eq) in enumerate(eqs1)
+        for (j, eq2) in enumerate(eqs2)
+            if isequal(eq, eq2)
+                push!(eqpairs, i => j)
+                break
+            elseif !isequal(eq, eq2) && j == length(eqs2)
```
Comment (YingboMa) [inline src/systems/diffeqs/abstractodesystem.jl:917]: Why checking this at all if you don't use it?

File: src/systems/diffeqs/odesystem.jl [hunk 3]
```diff
@@ -367,7 +367,7 @@ function convert_system(::Type{<:ODESystem}, sys, t; name=nameof(sys))
                 newsts[i] = s
                 continue
             end
-            ns = operation(s)(t)
+            ns = operation(s)(t; metadata=SymbolicUtils.metadata(s))
```
Comment (YingboMa) [inline src/systems/diffeqs/odesystem.jl:370]: This should just be `similarterm`

**Review by YingboMa — CHANGES_REQUESTED** [review]
Checkout `similarterm`

File: src/systems/diffeqs/abstractodesystem.jl (hunk 2 above)
Comment (anandijain) [reply src/systems/diffeqs/abstractodesystem.jl]: for fun

**Review by YingboMa — APPROVED** [review]
LGTM

**Comment by YingboMa** [comment]
Test failure is unrelated.

**Comment by anandijain** [comment]
Shouldn't I have removed the Symbolics.substitute dispatch first though?

**Comment by YingboMa** [comment]
Ahhhh, yeah. Please do that.

**Pull request merged**